### PR TITLE
mgr/nfs: adjust export creation CLI args, and allow rgw user exports

### PR DESCRIPTION
**Pull request by liewegas**
- for ``nfs export create rgw``, take either bucket or user or both.  If we get only the user, do an export for the user that shows buckets at the top level.  If bucket+user, then use that user instead of the bucket owner.  If bucket only, behavior is as before (use bucket owner as the user).
- adjust hte CLI arg position for ``nfs export create cephfs`` to put the fsname as the last positional argument.  This aligns the 2 export create commands so that cluster_id and pseudo_path both come first (which FWIW is more intutive anyway)
- adjust tests so that all callers use named arguments for all of these commands, so that they work both before and after the CLI positional argument order change.

Note that the change for the ``nfs export create cephfs`` could be dropped, but I think it si worth the potential user breakage to end up with a better CLI.

TODO
- [x] add release note, and encourage users to use named arguments whenever using these interfaces
- [x] add test for rgw user-level export (including bucket creation with mkdir)

## Checklist
- [x] References tracker ticket
- [x] Updates documentation if necessary
- [x] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by liewegas** [comment]
jenkins test make check

**Comment by liewegas** [comment]
jenkins test make check

**Comment by liewegas** [comment]
@sebastian-philipp @alfonsomthd This is passing tests now!

File: src/pybind/mgr/nfs/export.py [hunk 1]
```diff
@@ -205,16 +205,19 @@ def _create_export_user(self, export: Export) -> None:
...
+                )
+                if ret:
+                    raise NFSException(f'Failed to fetch owner for bucket {export.path}')
+                j = json.loads(out)
+                owner = j.get('owner', '')
+                rgwfsal.user_id = owner
+            assert rgwfsal.user_id
             ret, out, err = self.mgr.tool_exec([
-                'radosgw-admin', 'user', 'info', '--uid', owner
+                'radosgw-admin', 'user', 'info', '--uid', rgwfsal.user_id
```
Comment (mgfritch) [inline src/pybind/mgr/nfs/export.py:220]: when a user_id that does not exist has been provided:
```
  File "/home/ceph/src/pybind/mgr/nfs/export.py", line 224, in _create_export_user
    f'Failed to fetch key for bucket {export.path} owner {owner}'
UnboundLocalError: local variable 'owner' referenced before assignment
```

**Review by mgfritch — APPROVED** [review]
one minor issue in the error path, but otherwise lgtm!

**Review by ajarr — APPROVED** [review]
Looks good!

**Review by alfonsomthd — APPROVED** [review]

File: src/pybind/mgr/nfs/tests/test_nfs.py [hunk 2]
```diff
@@ -163,6 +163,12 @@ def _mock_orchestrator(self, enable: bool) -> Iterator:
         ] if enable else []
 
         def mock_exec(cls, args):
+            #print(f"exec: {args}")
```
Comment (sebastian-philipp) [inline src/pybind/mgr/nfs/tests/test_nfs.py:166]: nit

**Review by sebastian-philipp — APPROVED** [review]

**Pull request merged**